### PR TITLE
Convert amounts to satoshis for internal operations.

### PR DESCRIPTION
**Pull request by piggydoughnut**
There was an error thrown when sending transaction with decimal number value.

### CODE REVIEW

File: src/util/helpers.ts [hunk 1]
```diff
@@ -28,6 +29,15 @@ export const formatDec = (num: number) => {
   return Number(num.toFixed(Config.DECIMAL)).toString();
 };
 
+export const toBitcoinAmount = (amount: number | string) => {
+  const txAmount = typeof amount !== 'string' ? amount.toString() : amount;
```
Comment (lenilsonjr) [inline src/util/helpers.ts:33]: maybe this can be shortened to 
```suggestion
  const txAmount = String(amount);
```

File: src/util/helpers.ts [hunk 2]
```diff
@@ -28,6 +29,15 @@ export const formatDec = (num: number) => {
   return Number(num.toFixed(Config.DECIMAL)).toString();
 };
 
+export const toBitcoinAmount = (amount: number | string) => {
```
Comment (lenilsonjr) [inline src/util/helpers.ts:32]: you can type the function return so we know what to expect: 
```suggestion
export const toBitcoinAmount = (amount: number | string): string => {
```

File: src/util/helpers.ts [hunk 3]
```diff
@@ -28,6 +29,15 @@ export const formatDec = (num: number) => {
...
+  const txAmount = typeof amount !== 'string' ? amount.toString() : amount;
+  const value = toBitcoin(txAmount);
+  return Number(value.toFixed(Config.DECIMAL)).toString();
+};
+
+export const getUsdValue = (amount: number, tokelPriceUSD: number) =>
```
Comment (lenilsonjr) [inline src/util/helpers.ts:38]: I'd love if `amount` (in the function above as well) could be named `amountInSatoshis` as to make really explicit what we're handling.

File: src/components/Dashboard/widgets/Wallet/TxConfirmation.tsx [hunk 4]
```diff
@@ -53,7 +54,7 @@ const TxConfirmation = ({
       )}
       {txId && (
         <TxInformation
-          amount={amount}
+          amount={toSatoshi(amount)}
```
Comment (lenilsonjr) [inline src/components/Dashboard/widgets/Wallet/TxConfirmation.tsx:57]: this prop could also be named `amountInSatoshi` for explicitness.

File: src/components/Dashboard/widgets/Wallet/TxInformation.tsx [hunk 5]
```diff
@@ -56,6 +56,7 @@ const TxInformation = ({
 }: TxConfirmationProps): ReactElement => {
   const tokenTx = useSelector(selectCurrenTxTokenTx);
   const currentToken = useSelector(selectCurrentTokenInfo);
+  const txAmount = toBitcoinAmount(amount);
   // const usdValueTemp = formatFiat(Number(amount) * Number(usdValue));
```
Comment (lenilsonjr) [inline src/components/Dashboard/widgets/Wallet/TxInformation.tsx:60]: clean up?

File: src/util/transactions.ts [hunk 6]
```diff
@@ -24,15 +21,16 @@ export const parseBlockchainTransaction = (tx, address: string) => {
  * @returns
  */
 export const parseSpendTx = (newtx, from) => {
+  console.log('parseSpendTx ', newtx);
```
Comment (lenilsonjr) [inline src/util/transactions.ts:24]: this is log intended to stay or was added here as temporary debug? (_this part of a bigger question I have of how we should approach logging, what makes sense to log etc._

**Review by lenilsonjr — APPROVED** [review]
looks good, some comments for consideration!

File: src/util/transactions.ts (hunk 6 above)
Comment (piggydoughnut) [reply src/util/transactions.ts:24]: it is a whoopsie

File: src/components/Dashboard/widgets/Wallet/TxInformation.tsx (hunk 5 above)
Comment (piggydoughnut) [reply src/components/Dashboard/widgets/Wallet/TxInformation.tsx:60]: I guess I could get rid of those, these are commented out coz we have usd value in the designs, but we used to have no USD value as we werent tracked anywhere. It doesnt look like anyone is missing it though. I ll remove it.

**Pull request merged**